src/wxProfilerGUI/profilergui.cpp

```diff
@@ -642,25 +642,15 @@ bool ProfilerGUI::Run()
 	{
 		std::unique_ptr<AttachInfo> info(AttachToProcess(cmdline_attach));
 		if (!cmdline_thread_ids.empty()) {
-			std::cout << "(Orig) Profiling Threads: ";
-			for (auto hwnd : info->thread_handles)
-				std::cout << " " << (uintptr_t)hwnd;
 			std::vector<HANDLE> profile_threads;
 			for (auto tid_h : info->thread_handles) {
 				DWORD test_tid = GetThreadId(tid_h);
-				std::cout << "Checking thread handle: " << (uintptr_t)tid_h << "/" << test_tid << "\n";
-				if (std::find(cmdline_thread_ids.begin(),cmdline_thread_ids.end(),test_tid)
-					!= cmdline_thread_ids.end()) {
-					std::cout << "  Match!\n";
+				if (std::find(cmdline_thread_ids.begin(),cmdline_thread_ids.end(),test_tid) != cmdline_thread_ids.end()) {
 					profile_threads.push_back(tid_h);
 				}
 			}
 			info->thread_handles = profile_threads;
-			std::cout << "Profiling Threads: ";
-			for (auto hwnd : info->thread_handles)
-				std::cout << " " << (uintptr_t)hwnd;
-			std::cout << "\n";
-
+			// Is there a threshold we should use to set this? (currently: attach_count < total_count)

```

Comment:
I think attaching to select threads and also to new threads would make sense as something explicitly opt-in. In any case, please don't leave open-ended questions in comments in the source code :)

Reply:
It was more of a question to someone more knowledgeable about the code than I. I can remove it.

Reply:
In `ProfilerGUI::LaunchProfiler(...)`, `attach_all_threads` is used to decide whether or not to launch a new `Debugger` thread. This change (sort of) adds a wrinkle. There could be a number of threads to profile without being _all_ of the threads.

Under what conditions do we want to launch the other debugger thread? The comments in the code don't make this entirely clear. I assume it's for performance reasons, but I don't really know when doing this is worth it.

Reply:
You can already select multiple threads with the GUI, and so I attempted to retain what was probably the existing behavior, WRT `attach_all_threads`. It just seemed like someone ought to make an intelligent decision about this.